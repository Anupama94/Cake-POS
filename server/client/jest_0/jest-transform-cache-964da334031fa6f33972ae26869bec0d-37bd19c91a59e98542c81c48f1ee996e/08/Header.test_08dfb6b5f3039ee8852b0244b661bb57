2d7d74eaa5b6eaa652e38aa347cc4230
"use strict";

var _interopRequireDefault = require("/home/anupamaa/Desktop/Training/pos/server/client/node_modules/@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _enzyme = require("enzyme");

var _Header = _interopRequireDefault(require("./Header"));

var _enzymeToJson = require("enzyme-to-json");

var _jsxFileName = "/home/anupamaa/Desktop/Training/pos/server/client/src/components/Shared/Header.test.js";
describe("<Header />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Header.default, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 13
      },
      __self: void 0
    }));
  });
  it("should render correctly", () => {
    expect((0, _enzymeToJson.shallowToJson)(wrapper)).toMatchSnapshot();
  });
  it('includes an h2 tag that specifies the application name', () => {
    expect(wrapper.find('h2.orangify').text()).toBe("Cake | POS");
  }); // it("should handle the click event", () => {
  //     const logOutSpy = jest.spyOn(wrapper.instance(), "logout");
  //     const context = {
  //         history : {
  //             push: jest.fn()
  //         }
  //     }
  //     const logOutAuthSpy = jest.fn();
  //
  //     window.confirm = jest.fn(() => true);
  //     wrapper.instance().forceUpdate();
  //     const logoutButton = wrapper.find('#logout-btn');
  //     expect(logOutSpy).not.toHaveBeenCalled();
  //     logoutButton.simulate('click');
  //     expect(logOutSpy).toHaveBeenCalledTimes(1);
  //     expect(window.confirm).toBeCalled();
  // })
  // it('includes a Button to logout', () => {
  //     const logoutButton = wrapper.find('Button').text().toBeTruthy;
  //     logoutButton.simulate('click');
  // });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImRlc2NyaWJlIiwid3JhcHBlciIsImJlZm9yZUVhY2giLCJpdCIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCIsImZpbmQiLCJ0ZXh0IiwidG9CZSJdLCJtYXBwaW5ncyI6Ijs7OztBQUFBOztBQUNBOztBQUVBOztBQUNBOztBQUNBOzs7QUFHQUEsUUFBUSxDQUFDLFlBQUQsRUFBZSxNQUFNO0FBRXpCLE1BQUlDLE9BQUo7QUFFQUMsRUFBQUEsVUFBVSxDQUFDLE1BQU07QUFBRUQsSUFBQUEsT0FBTyxHQUFHLHFCQUFRLDZCQUFDLGVBQUQ7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUEsTUFBUixDQUFWO0FBQWdDLEdBQXpDLENBQVY7QUFFQUUsRUFBQUEsRUFBRSxDQUFDLHlCQUFELEVBQTRCLE1BQU07QUFDaENDLElBQUFBLE1BQU0sQ0FBQyxpQ0FBY0gsT0FBZCxDQUFELENBQU4sQ0FBK0JJLGVBQS9CO0FBQ0gsR0FGQyxDQUFGO0FBS0FGLEVBQUFBLEVBQUUsQ0FBQyx3REFBRCxFQUEyRCxNQUFNO0FBRS9EQyxJQUFBQSxNQUFNLENBQUNILE9BQU8sQ0FBQ0ssSUFBUixDQUFhLGFBQWIsRUFBNEJDLElBQTVCLEVBQUQsQ0FBTixDQUEyQ0MsSUFBM0MsQ0FBZ0QsWUFBaEQ7QUFDSCxHQUhDLENBQUYsQ0FYeUIsQ0FnQnpCO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFHQTtBQUNBO0FBQ0E7QUFDQTtBQUVILENBeENPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IFJlYWN0RE9NIGZyb20gJ3JlYWN0LWRvbSc7XG5cbmltcG9ydCB7IHNoYWxsb3cgfSBmcm9tICdlbnp5bWUnO1xuaW1wb3J0IEhlYWRlciBmcm9tICcuL0hlYWRlcic7XG5pbXBvcnQgeyBzaGFsbG93VG9Kc29uIH0gZnJvbSAnZW56eW1lLXRvLWpzb24nO1xuXG5cbmRlc2NyaWJlKFwiPEhlYWRlciAvPlwiLCAoKSA9PiB7XG5cbiAgICBsZXQgd3JhcHBlcjtcblxuICAgIGJlZm9yZUVhY2goKCkgPT4geyB3cmFwcGVyID0gc2hhbGxvdyg8SGVhZGVyIC8+KTsgfSk7XG5cbiAgICBpdChcInNob3VsZCByZW5kZXIgY29ycmVjdGx5XCIsICgpID0+IHtcbiAgICAgICAgZXhwZWN0KHNoYWxsb3dUb0pzb24od3JhcHBlcikpLnRvTWF0Y2hTbmFwc2hvdCgpO1xuICAgIH0pO1xuXG5cbiAgICBpdCgnaW5jbHVkZXMgYW4gaDIgdGFnIHRoYXQgc3BlY2lmaWVzIHRoZSBhcHBsaWNhdGlvbiBuYW1lJywgKCkgPT4ge1xuXG4gICAgICAgIGV4cGVjdCh3cmFwcGVyLmZpbmQoJ2gyLm9yYW5naWZ5JykudGV4dCgpKS50b0JlKFwiQ2FrZSB8IFBPU1wiKTtcbiAgICB9KTtcblxuICAgIC8vIGl0KFwic2hvdWxkIGhhbmRsZSB0aGUgY2xpY2sgZXZlbnRcIiwgKCkgPT4ge1xuICAgIC8vICAgICBjb25zdCBsb2dPdXRTcHkgPSBqZXN0LnNweU9uKHdyYXBwZXIuaW5zdGFuY2UoKSwgXCJsb2dvdXRcIik7XG4gICAgLy8gICAgIGNvbnN0IGNvbnRleHQgPSB7XG4gICAgLy8gICAgICAgICBoaXN0b3J5IDoge1xuICAgIC8vICAgICAgICAgICAgIHB1c2g6IGplc3QuZm4oKVxuICAgIC8vICAgICAgICAgfVxuICAgIC8vICAgICB9XG4gICAgLy8gICAgIGNvbnN0IGxvZ091dEF1dGhTcHkgPSBqZXN0LmZuKCk7XG4gICAgLy9cbiAgICAvLyAgICAgd2luZG93LmNvbmZpcm0gPSBqZXN0LmZuKCgpID0+IHRydWUpO1xuICAgIC8vICAgICB3cmFwcGVyLmluc3RhbmNlKCkuZm9yY2VVcGRhdGUoKTtcbiAgICAvLyAgICAgY29uc3QgbG9nb3V0QnV0dG9uID0gd3JhcHBlci5maW5kKCcjbG9nb3V0LWJ0bicpO1xuICAgIC8vICAgICBleHBlY3QobG9nT3V0U3B5KS5ub3QudG9IYXZlQmVlbkNhbGxlZCgpO1xuICAgIC8vICAgICBsb2dvdXRCdXR0b24uc2ltdWxhdGUoJ2NsaWNrJyk7XG4gICAgLy8gICAgIGV4cGVjdChsb2dPdXRTcHkpLnRvSGF2ZUJlZW5DYWxsZWRUaW1lcygxKTtcbiAgICAvLyAgICAgZXhwZWN0KHdpbmRvdy5jb25maXJtKS50b0JlQ2FsbGVkKCk7XG4gICAgLy8gfSlcblxuXG4gICAgLy8gaXQoJ2luY2x1ZGVzIGEgQnV0dG9uIHRvIGxvZ291dCcsICgpID0+IHtcbiAgICAvLyAgICAgY29uc3QgbG9nb3V0QnV0dG9uID0gd3JhcHBlci5maW5kKCdCdXR0b24nKS50ZXh0KCkudG9CZVRydXRoeTtcbiAgICAvLyAgICAgbG9nb3V0QnV0dG9uLnNpbXVsYXRlKCdjbGljaycpO1xuICAgIC8vIH0pO1xuXG59KTtcblxuXG4iXX0=