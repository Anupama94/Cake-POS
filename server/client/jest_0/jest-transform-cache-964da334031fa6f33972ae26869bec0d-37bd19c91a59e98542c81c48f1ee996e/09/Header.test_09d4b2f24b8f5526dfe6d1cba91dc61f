5ce311e036c9bc28607ffee9f5de2f81
"use strict";

var _interopRequireDefault = require("/home/anupamaa/Desktop/Training/pos/server/client/node_modules/@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _enzymeAdapterReact = _interopRequireDefault(require("enzyme-adapter-react-16"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _Header = _interopRequireDefault(require("./Header"));

var _jsxFileName = "/home/anupamaa/Desktop/Training/pos/server/client/src/components/Shared/Header.test.js";
;
(0, _enzyme.configure)({
  adapter: new _enzymeAdapterReact.default()
});
describe("<Header />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Header.default, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 15
      },
      __self: void 0
    }));
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');

    _reactDom.default.render(_react.default.createElement(_Header.default, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 21
      },
      __self: void 0
    }), div);

    _reactDom.default.unmountComponentAtNode(div);
  });
  it('includes an h2 tag that specifies the application name', () => {
    expect(wrapper.find('h2.orangify').text()).toBe("Cake | POS");
  });
  it('includes a nav-bar-brand that specifies the application name', () => {
    expect(wrapper.find('Button').text()).toBe("Cake | POS");
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImFkYXB0ZXIiLCJBZGFwdGVyIiwiZGVzY3JpYmUiLCJ3cmFwcGVyIiwiYmVmb3JlRWFjaCIsIml0IiwiZGl2IiwiZG9jdW1lbnQiLCJjcmVhdGVFbGVtZW50IiwiUmVhY3RET00iLCJyZW5kZXIiLCJ1bm1vdW50Q29tcG9uZW50QXROb2RlIiwiZXhwZWN0IiwiZmluZCIsInRleHQiLCJ0b0JlIl0sIm1hcHBpbmdzIjoiOzs7O0FBQUE7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBR0E7OztBQUhpQztBQUtqQyx1QkFBVTtBQUFDQSxFQUFBQSxPQUFPLEVBQUUsSUFBSUMsMkJBQUo7QUFBVixDQUFWO0FBRUFDLFFBQVEsQ0FBQyxZQUFELEVBQWUsTUFBTTtBQUV6QixNQUFJQyxPQUFKO0FBRUFDLEVBQUFBLFVBQVUsQ0FBQyxNQUFNO0FBQUVELElBQUFBLE9BQU8sR0FBRyxxQkFBUSw2QkFBQyxlQUFEO0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBLE1BQVIsQ0FBVjtBQUFnQyxHQUF6QyxDQUFWO0FBR0FFLEVBQUFBLEVBQUUsQ0FBQywwQkFBRCxFQUE2QixNQUFNO0FBRWpDLFVBQU1DLEdBQUcsR0FBR0MsUUFBUSxDQUFDQyxhQUFULENBQXVCLEtBQXZCLENBQVo7O0FBQ0FDLHNCQUFTQyxNQUFULENBQWdCLDZCQUFDLGVBQUQ7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUEsTUFBaEIsRUFBNEJKLEdBQTVCOztBQUNBRyxzQkFBU0Usc0JBQVQsQ0FBZ0NMLEdBQWhDO0FBQ0gsR0FMQyxDQUFGO0FBUUFELEVBQUFBLEVBQUUsQ0FBQyx3REFBRCxFQUEyRCxNQUFNO0FBRS9ETyxJQUFBQSxNQUFNLENBQUNULE9BQU8sQ0FBQ1UsSUFBUixDQUFhLGFBQWIsRUFBNEJDLElBQTVCLEVBQUQsQ0FBTixDQUEyQ0MsSUFBM0MsQ0FBZ0QsWUFBaEQ7QUFDSCxHQUhDLENBQUY7QUFNQVYsRUFBQUEsRUFBRSxDQUFDLDhEQUFELEVBQWlFLE1BQU07QUFFckVPLElBQUFBLE1BQU0sQ0FBQ1QsT0FBTyxDQUFDVSxJQUFSLENBQWEsUUFBYixFQUF1QkMsSUFBdkIsRUFBRCxDQUFOLENBQXNDQyxJQUF0QyxDQUEyQyxZQUEzQztBQUNILEdBSEMsQ0FBRjtBQUtILENBMUJPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IHsgY29uZmlndXJlIH0gZnJvbSAnZW56eW1lJztcbmltcG9ydCBBZGFwdGVyIGZyb20gJ2VuenltZS1hZGFwdGVyLXJlYWN0LTE2J1xuaW1wb3J0IFJlYWN0RE9NIGZyb20gJ3JlYWN0LWRvbSc7O1xuXG5pbXBvcnQgeyBzaGFsbG93IH0gZnJvbSAnZW56eW1lJztcbmltcG9ydCBIZWFkZXIgZnJvbSAnLi9IZWFkZXInO1xuXG5jb25maWd1cmUoe2FkYXB0ZXI6IG5ldyBBZGFwdGVyKCl9KTtcblxuZGVzY3JpYmUoXCI8SGVhZGVyIC8+XCIsICgpID0+IHtcblxuICAgIGxldCB3cmFwcGVyO1xuXG4gICAgYmVmb3JlRWFjaCgoKSA9PiB7IHdyYXBwZXIgPSBzaGFsbG93KDxIZWFkZXIgLz4pOyB9KTtcblxuXG4gICAgaXQoJ3JlbmRlcnMgd2l0aG91dCBjcmFzaGluZycsICgpID0+IHtcblxuICAgICAgICBjb25zdCBkaXYgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdkaXYnKTtcbiAgICAgICAgUmVhY3RET00ucmVuZGVyKDxIZWFkZXIgLz4sIGRpdik7XG4gICAgICAgIFJlYWN0RE9NLnVubW91bnRDb21wb25lbnRBdE5vZGUoZGl2KTtcbiAgICB9KTtcblxuXG4gICAgaXQoJ2luY2x1ZGVzIGFuIGgyIHRhZyB0aGF0IHNwZWNpZmllcyB0aGUgYXBwbGljYXRpb24gbmFtZScsICgpID0+IHtcblxuICAgICAgICBleHBlY3Qod3JhcHBlci5maW5kKCdoMi5vcmFuZ2lmeScpLnRleHQoKSkudG9CZShcIkNha2UgfCBQT1NcIik7XG4gICAgfSk7XG5cblxuICAgIGl0KCdpbmNsdWRlcyBhIG5hdi1iYXItYnJhbmQgdGhhdCBzcGVjaWZpZXMgdGhlIGFwcGxpY2F0aW9uIG5hbWUnLCAoKSA9PiB7XG5cbiAgICAgICAgZXhwZWN0KHdyYXBwZXIuZmluZCgnQnV0dG9uJykudGV4dCgpKS50b0JlKFwiQ2FrZSB8IFBPU1wiKTtcbiAgICB9KTtcblxufSk7Il19