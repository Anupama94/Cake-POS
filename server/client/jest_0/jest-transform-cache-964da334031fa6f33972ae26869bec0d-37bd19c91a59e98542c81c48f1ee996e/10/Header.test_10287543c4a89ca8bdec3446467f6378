e1578327356e5738554d2fabafce0333
"use strict";

var _interopRequireDefault = require("/home/anupamaa/Desktop/Training/pos/server/client/node_modules/@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _enzymeAdapterReact = _interopRequireDefault(require("enzyme-adapter-react-16"));

var _Header = _interopRequireDefault(require("./Header"));

var _jsxFileName = "/home/anupamaa/Desktop/Training/pos/server/client/src/components/Shared/Header.test.js";
(0, _enzyme.configure)({
  adapter: new _enzymeAdapterReact.default()
});
describe("<Header />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = (0, _enzyme.shallow)(_react.default.createElement("div", {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 14
      },
      __self: void 0
    }, _react.default.createElement("button", {
      className: "btn btn-primary",
      __source: {
        fileName: _jsxFileName,
        lineNumber: 14
      },
      __self: void 0
    }, "OK")));
  });
  it('includes a nav-bar-brand that specifies the application name', () => {
    const button = wrapper.find('.btn');
    expect(button.text()).toBe('OK');
  }); // it('should render a anchor tag', () => {
  //     const wrapper = shallow(
  //         <a
  //     className="App-link"
  //     href="https://reactjs.org"
  //     target="_blank"
  //     rel="noopener noreferrer"
  //         >
  //         Learn React
  //     </a>
  // );
  //
  //     // expect(component).toMatchSnapshot();
  //     expect(wrapper).toMatchSnapshot();
  // });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImFkYXB0ZXIiLCJBZGFwdGVyIiwiZGVzY3JpYmUiLCJ3cmFwcGVyIiwiYmVmb3JlRWFjaCIsIml0IiwiYnV0dG9uIiwiZmluZCIsImV4cGVjdCIsInRleHQiLCJ0b0JlIl0sIm1hcHBpbmdzIjoiOzs7O0FBQUE7O0FBQ0E7O0FBQ0E7O0FBR0E7OztBQUVBLHVCQUFVO0FBQUNBLEVBQUFBLE9BQU8sRUFBRSxJQUFJQywyQkFBSjtBQUFWLENBQVY7QUFFQUMsUUFBUSxDQUFDLFlBQUQsRUFBZSxNQUFNO0FBRXpCLE1BQUlDLE9BQUo7QUFFQUMsRUFBQUEsVUFBVSxDQUFDLE1BQU07QUFBQ0QsSUFBQUEsT0FBTyxHQUFHLHFCQUFRO0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBLE9BQUs7QUFBUSxNQUFBLFNBQVMsRUFBQyxpQkFBbEI7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUEsWUFBTCxDQUFSLENBQVY7QUFBZ0YsR0FBeEYsQ0FBVjtBQUVBRSxFQUFBQSxFQUFFLENBQUMsOERBQUQsRUFBaUUsTUFBTTtBQUVyRSxVQUFNQyxNQUFNLEdBQUdILE9BQU8sQ0FBQ0ksSUFBUixDQUFhLE1BQWIsQ0FBZjtBQUNBQyxJQUFBQSxNQUFNLENBQUNGLE1BQU0sQ0FBQ0csSUFBUCxFQUFELENBQU4sQ0FBc0JDLElBQXRCLENBQTJCLElBQTNCO0FBQ0gsR0FKQyxDQUFGLENBTnlCLENBWXpCO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNILENBM0JPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IHsgY29uZmlndXJlIH0gZnJvbSAnZW56eW1lJztcbmltcG9ydCBBZGFwdGVyIGZyb20gJ2VuenltZS1hZGFwdGVyLXJlYWN0LTE2JztcblxuaW1wb3J0IHsgc2hhbGxvdyB9IGZyb20gJ2VuenltZSc7XG5pbXBvcnQgSGVhZGVyIGZyb20gJy4vSGVhZGVyJztcblxuY29uZmlndXJlKHthZGFwdGVyOiBuZXcgQWRhcHRlcigpfSk7XG5cbmRlc2NyaWJlKFwiPEhlYWRlciAvPlwiLCAoKSA9PiB7XG5cbiAgICBsZXQgd3JhcHBlcjtcblxuICAgIGJlZm9yZUVhY2goKCkgPT4ge3dyYXBwZXIgPSBzaGFsbG93KDxkaXY+PGJ1dHRvbiBjbGFzc05hbWU9J2J0biBidG4tcHJpbWFyeSc+T0s8L2J1dHRvbj48L2Rpdj4pOyB9KTtcblxuICAgIGl0KCdpbmNsdWRlcyBhIG5hdi1iYXItYnJhbmQgdGhhdCBzcGVjaWZpZXMgdGhlIGFwcGxpY2F0aW9uIG5hbWUnLCAoKSA9PiB7XG5cbiAgICAgICAgY29uc3QgYnV0dG9uID0gd3JhcHBlci5maW5kKCcuYnRuJyk7XG4gICAgICAgIGV4cGVjdChidXR0b24udGV4dCgpKS50b0JlKCdPSycpO1xuICAgIH0pO1xuXG4gICAgLy8gaXQoJ3Nob3VsZCByZW5kZXIgYSBhbmNob3IgdGFnJywgKCkgPT4ge1xuICAgIC8vICAgICBjb25zdCB3cmFwcGVyID0gc2hhbGxvdyhcbiAgICAvLyAgICAgICAgIDxhXG4gICAgLy8gICAgIGNsYXNzTmFtZT1cIkFwcC1saW5rXCJcbiAgICAvLyAgICAgaHJlZj1cImh0dHBzOi8vcmVhY3Rqcy5vcmdcIlxuICAgIC8vICAgICB0YXJnZXQ9XCJfYmxhbmtcIlxuICAgIC8vICAgICByZWw9XCJub29wZW5lciBub3JlZmVycmVyXCJcbiAgICAvLyAgICAgICAgID5cbiAgICAvLyAgICAgICAgIExlYXJuIFJlYWN0XG4gICAgLy8gICAgIDwvYT5cbiAgICAvLyApO1xuICAgIC8vXG4gICAgLy8gICAgIC8vIGV4cGVjdChjb21wb25lbnQpLnRvTWF0Y2hTbmFwc2hvdCgpO1xuICAgIC8vICAgICBleHBlY3Qod3JhcHBlcikudG9NYXRjaFNuYXBzaG90KCk7XG4gICAgLy8gfSk7XG59KTsiXX0=