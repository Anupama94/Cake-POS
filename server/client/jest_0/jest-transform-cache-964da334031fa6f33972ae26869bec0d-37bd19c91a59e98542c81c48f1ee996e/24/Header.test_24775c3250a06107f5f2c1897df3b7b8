46164155233c72c1bdf6ebba25b0099e
"use strict";

var _interopRequireDefault = require("/home/anupamaa/Desktop/Training/pos/server/client/node_modules/@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _enzyme = require("enzyme");

var _Header = _interopRequireDefault(require("./Header"));

var _jsxFileName = "/home/anupamaa/Desktop/Training/pos/server/client/src/components/Shared/Header.test.js";
describe("<Header />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Header.default, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 12
      },
      __self: void 0
    }));
  });
  it('includes an h2 tag that specifies the application name', () => {
    expect(wrapper.find('h2.orangify').text()).toBe("Cake | POS");
  });
  it('includes a Button to logout', () => {
    expect(wrapper.find('Button').text()).toBeTruthy;
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImRlc2NyaWJlIiwid3JhcHBlciIsImJlZm9yZUVhY2giLCJpdCIsImV4cGVjdCIsImZpbmQiLCJ0ZXh0IiwidG9CZSIsInRvQmVUcnV0aHkiXSwibWFwcGluZ3MiOiI7Ozs7QUFBQTs7QUFDQTs7QUFFQTs7QUFDQTs7O0FBR0FBLFFBQVEsQ0FBQyxZQUFELEVBQWUsTUFBTTtBQUV6QixNQUFJQyxPQUFKO0FBRUFDLEVBQUFBLFVBQVUsQ0FBQyxNQUFNO0FBQUVELElBQUFBLE9BQU8sR0FBRyxxQkFBUSw2QkFBQyxlQUFEO0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBLE1BQVIsQ0FBVjtBQUFnQyxHQUF6QyxDQUFWO0FBS0FFLEVBQUFBLEVBQUUsQ0FBQyx3REFBRCxFQUEyRCxNQUFNO0FBRS9EQyxJQUFBQSxNQUFNLENBQUNILE9BQU8sQ0FBQ0ksSUFBUixDQUFhLGFBQWIsRUFBNEJDLElBQTVCLEVBQUQsQ0FBTixDQUEyQ0MsSUFBM0MsQ0FBZ0QsWUFBaEQ7QUFDSCxHQUhDLENBQUY7QUFNQUosRUFBQUEsRUFBRSxDQUFDLDZCQUFELEVBQWdDLE1BQU07QUFFcENDLElBQUFBLE1BQU0sQ0FBQ0gsT0FBTyxDQUFDSSxJQUFSLENBQWEsUUFBYixFQUF1QkMsSUFBdkIsRUFBRCxDQUFOLENBQXNDRSxVQUF0QztBQUNILEdBSEMsQ0FBRjtBQUtILENBcEJPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IFJlYWN0RE9NIGZyb20gJ3JlYWN0LWRvbSc7XG5cbmltcG9ydCB7IHNoYWxsb3cgfSBmcm9tICdlbnp5bWUnO1xuaW1wb3J0IEhlYWRlciBmcm9tICcuL0hlYWRlcic7XG5cblxuZGVzY3JpYmUoXCI8SGVhZGVyIC8+XCIsICgpID0+IHtcblxuICAgIGxldCB3cmFwcGVyO1xuXG4gICAgYmVmb3JlRWFjaCgoKSA9PiB7IHdyYXBwZXIgPSBzaGFsbG93KDxIZWFkZXIgLz4pOyB9KTtcblxuXG5cblxuICAgIGl0KCdpbmNsdWRlcyBhbiBoMiB0YWcgdGhhdCBzcGVjaWZpZXMgdGhlIGFwcGxpY2F0aW9uIG5hbWUnLCAoKSA9PiB7XG5cbiAgICAgICAgZXhwZWN0KHdyYXBwZXIuZmluZCgnaDIub3JhbmdpZnknKS50ZXh0KCkpLnRvQmUoXCJDYWtlIHwgUE9TXCIpO1xuICAgIH0pO1xuXG5cbiAgICBpdCgnaW5jbHVkZXMgYSBCdXR0b24gdG8gbG9nb3V0JywgKCkgPT4ge1xuXG4gICAgICAgIGV4cGVjdCh3cmFwcGVyLmZpbmQoJ0J1dHRvbicpLnRleHQoKSkudG9CZVRydXRoeTtcbiAgICB9KTtcblxufSk7Il19