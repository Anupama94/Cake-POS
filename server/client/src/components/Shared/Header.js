import React from 'react';
import { Nav, NavItem, Navbar, NavbarBrand, Collapse, NavbarToggler, Button } from 'reactstrap';
import PropTypes from 'prop-types';
import auth from '../../authentication/auth';


class Header extends React.Component {

    static contextTypes = {
        router: PropTypes.object
    }

    logout(e){

            if (window.confirm('Are you sure you wish to delete this item?')){
            auth.logout(() => {
                this.context.router.history.push("/");
            });
        }

    }

    render() {

        return (
            <div>
        <Navbar color="navbar-dark bg-dark" light expand="md">
        <NavbarBrand href="/"><h2 className="orangify">Cake | POS</h2></NavbarBrand>

    <Nav className="ml-auto" navbar>
    <NavItem>
    <Button className="orangify" id="logout-btn" onClick={this.logout.bind(this)} >Log Out</Button>
    </NavItem>
    </Nav>


    </Navbar>
    <br />
    </div>
    );
    }
}

export default Header;