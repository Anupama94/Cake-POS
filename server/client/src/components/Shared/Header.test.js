import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import Header from './Header';
import { shallowToJson } from 'enzyme-to-json';


describe("<Header />", () => {

    let wrapper;

    beforeEach(() => {
         wrapper = shallow(<Header />); 
    });

    it("should render correctly", () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });


    it('includes an h2 tag that specifies the application name', () => {

        expect(wrapper.find('h2.orangify').text()).toBe("Cake | POS");
    });

    // it("should handle the click event", () => {
    //     const logOutSpy = jest.spyOn(wrapper.instance(), "logout");
    //     const context = {
    //         history : {
    //             push: jest.fn()
    //         }
    //     }
    //     const logOutAuthSpy = jest.fn();
    //
    //     window.confirm = jest.fn(() => true);
    //     wrapper.instance().forceUpdate();
    //     const logoutButton = wrapper.find('#logout-btn');
    //     expect(logOutSpy).not.toHaveBeenCalled();
    //     logoutButton.simulate('click');
    //     expect(logOutSpy).toHaveBeenCalledTimes(1);
    //     expect(window.confirm).toBeCalled();
    // })


    // it('includes a Button to logout', () => {
    //     const logoutButton = wrapper.find('Button').text().toBeTruthy;
    //     logoutButton.simulate('click');
    // });

});


