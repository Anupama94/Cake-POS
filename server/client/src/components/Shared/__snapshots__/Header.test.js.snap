// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> includes a nav-bar-brand that specifies the application name 1`] = `
<Navbar
  color="navbar-dark bg-dark"
  expand="md"
  light={true}
  tag="nav"
>
  <NavbarBrand
    href="/"
    tag="a"
  >
    <h2
      className="orangify"
    >
       Cake | POS 
    </h2>
  </NavbarBrand>
  <NavbarToggler
    tag="button"
    type="button"
  />
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    navbar={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <Button
          color="secondary"
          onClick={[Function]}
          style={
            Object {
              "color": "orange",
            }
          }
          tag="button"
        >
          Log Out
        </Button>
      </NavItem>
    </Nav>
  </Collapse>
</Navbar>
`;

exports[`<Header /> should render a anchor tag 1`] = `
<a
  className="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React
</a>
`;

exports[`<Header /> should render correctly 1`] = `
<div>
  <Navbar
    color="navbar-dark bg-dark"
    expand="md"
    light={true}
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      <h2
        className="orangify"
      >
        Cake | POS
      </h2>
    </NavbarBrand>
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <Button
          className="orangify"
          color="secondary"
          id="logout-btn"
          onClick={[Function]}
          tag="button"
        >
          Log Out
        </Button>
      </NavItem>
    </Nav>
  </Navbar>
  <br />
</div>
`;

exports[`App should render a anchor tag 1`] = `
<a
  className="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React
</a>
`;
